plugins/dynamix.docker.manager/include/CreateDocker.php

```diff
@@ -88,9 +88,7 @@ function cpu_pinning() {
       $oldXML = simplexml_load_file($filename);
       if ($oldXML->Icon != $_POST['contIcon']) {
         if (!strpos($Repository,":")) $Repository .= ":latest";
-        $iconPath = $DockerTemplates->getIcon($Repository,$Name);
-        @unlink("$docroot/$iconPath");
-        @unlink("{$dockerManPaths['images']}/".basename($iconPath));
+        $DockerTemplates->purgeUnusedIconFiles($Name);
```

Comment:
Since removing an icon from the cache is a bit more complex I use a new function specifically for that purpose.